pkgs/top-level/splice.nix

```diff
@@ -120,12 +140,28 @@ in
   # We use `callPackage' to be able to omit function arguments that can be
   # obtained `pkgs` or `buildPackages` and their `xorg` package sets. Use
   # `newScope' for sets of packages in `pkgs' (see e.g. `gnome' below).
-  callPackage = pkgs.newScope {};
+  callPackage = pkgs.newScope { };
 
   callPackages = lib.callPackagesWith splicedPackagesWithXorg;
 
   newScope = extra: lib.callPackageWith (splicedPackagesWithXorg // extra);
 
+  makeScopeWithSplicing = lib.makeScopeWithSplicing splicePackages pkgs.newScope;
+
+  topProjectedSpliced = attr:
```

Comment:
ideas for a better name?

Reply:
``topSplices``, ``topProjectedSplices`` or ``projectedSplices``?
I'm not sure ``top`` really needs to be there.

Reply:
renamed to generateSplicesForMkScope